src/main/java/org/jetbrains/bsp/bazel/server/bsp/resolvers/TargetsRulesResolver.java

```diff
@@ -0,0 +1,63 @@
+package org.jetbrains.bsp.bazel.server.bsp.resolvers;
+
+import ch.epfl.scala.bsp4j.BuildTargetIdentifier;
+import com.google.devtools.build.lib.query2.proto.proto2api.Build;
+import com.google.devtools.build.lib.query2.proto.proto2api.Build.Rule;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.jetbrains.bsp.bazel.server.bazel.BazelRunner;
+import org.jetbrains.bsp.bazel.server.bazel.data.BazelProcessResult;
+import org.jetbrains.bsp.bazel.server.bazel.params.BazelRunnerFlag;
+
+public class TargetsRulesResolver<T> {
```

Comment:
maybe TargetRulesResolver? It seems more natural

Reply:
also I'm not sure what's the point of this class, it seems to be a very simple wrapper only

Reply:
It is mentioned in the PR description as this will be useful for #22 and #25